File: parse_log.js
const fs = require('fs');
const path = require('path');

const logPath = path.join(__dirname, 'carro velho', 'log.txt');
const content = fs.readFileSync(logPath, 'utf8');

const matches = content.match(/>([0-9A-Fa-f]+)/g) || [];

const commands = new Set();
for (let m of matches) {
    m = m.substring(1); // remove '>'
    if (m.length >= 4) {
        const cmd = m.substring(0, 4);
        if (cmd.startsWith("01")) {
            commands.add(cmd);
        } else if (m.startsWith("22") && m.length >= 6) {
            commands.add(m.substring(0, 6));
        } else {
            commands.add(m);
        }
    }
}

console.log("Unique commands detected in ELM log:");
const sortedCmds = Array.from(commands).sort();
for (const c of sortedCmds) {
    let success = false;
    if (c.startsWith("01")) {
        const pid = c.substring(2, 4);
        const respPattern = "41" + pid;
        if (content.includes(respPattern)) {
            success = true;
        }
    } else if (c.startsWith("22")) {
        const udsPid = c.substring(2, 6);
        const respPattern = "62" + udsPid;
        if (content.includes(respPattern)) {
            success = true;
        }
    } else {
        success = "Unknown";
    }
    console.log(`Command: ${c} | Success: ${success}`);
}

File: parse_log.py
import re

log_path = r"C:\Users\Buia\.gemini\antigravity\scratch\gol_g1_dashboard\carro velho\log.txt"

with open(log_path, 'r') as f:
    content = f.read()

# Commands usually start with '>' followed by '01' (OBD Mode 1) or UDS commands
# Let's find all instances of '>' followed by alphanumeric characters
matches = re.findall(r'>([0-9A-Fa-f]+)', content)

commands = set()
for m in matches:
    # ELM commands are typically 4 hex chars for Mode 01 (e.g. 010C, 010D)
    # or they could be longer for UDS (e.g. 221154)
    # Let's see what lengths we have
    if len(m) >= 4:
        cmd = m[:4]
        if cmd.startswith("01"):
            commands.add(cmd)
        elif m.startswith("22") and len(m) >= 6:
            commands.add(m[:6])
        else:
            commands.add(m)

print("Unique commands detected in ELM log:")
for c in sorted(commands):
    # Check if there is a response in the log following this command
    # A typical response for Mode 01 PID XX looks like: 41 XX ...
    # We can check if "41" + PID exists in the log to confirm it succeeded
    pid = c[2:4] if c.startswith("01") else ""
    success = False
    if pid:
        resp_pattern = "41" + pid
        if resp_pattern in content:
            success = True
    elif c.startswith("22"):
        # UDS response usually starts with 62 + UDS PID
        uds_pid = c[2:6]
        resp_pattern = "62" + uds_pid
        if resp_pattern in content:
            success = True
    else:
        success = "Unknown"
        
    print(f"Command: {c} | Success: {success}")
